src/app/qgsmaptooldeletepart.cpp

```diff
@@ -108,6 +138,11 @@ void QgsMapToolDeletePart::canvasReleaseEvent( QgsMapMouseEvent *e )
   {
     emit messageEmitted( tr( "Couldn't remove the selected part." ) );
   }
+
+  if ( g.isEmpty() )
+  {
+    emit messageEmitted( tr( "All geometry parts deleted from feature with fid %1. Feature has no geometry now!" ).arg( mPressedFid ) );
```

Comment:
Wonder if "fid" can be confusing here, because of the default fid field in some formats. I suppose values won't necessarily match. Wouldn't "id" be clearer? Or feature %1?